src/main/java/com/hedvig/contract/graphql/resolvers/ContractQuery.kt

```diff
@@ -40,7 +41,18 @@ class ContractQuery(
   fun getInsuranceCost(env: DataFetchingEnvironment): InsuranceCost? {
     val memberId = env.getToken()
     val totalCost = premiumService.getTotalPremiumOfContractsWithActiveActiveInFutureOrPendingAgreementsByHolderMemberId(memberId)
-    val bundleDiscount = bundleDiscountService.getBundleDiscountOrNullForPeriod(memberId, YearMonth.now(), true)
+    //fixme: This is a lot of code in the wrong place but lets fix that by breaking out graphQL from contracts and campaign to it own module
+    val firstStartDate = contractService.getContractStatesByHolderMemberId(memberId)
+      .filterActivePendingOrActiveInFutureAgreements()
+      .filter { it.from != null }
+      .minBy { it.from!! }?.from
+
+    val periodToCheck = firstStartDate?.let {
+      val startMonth = YearMonth.from(it)
+      if (startMonth.isBefore(YearMonth.now())) YearMonth.now() else startMonth
+    } ?: YearMonth.now()
```

Comment:
What would happen if we just did "YearMonth.now()" straight up?

Reply:
I think we should use masterInception which is a set concept. Also a minor refactoring of the date calculation using Kotlin's `max`

```suggestion
    val firstMasterInceptionMaybe = contractService.getContractStatesByHolderMemberId(memberId)
      .filterActiveOrPendingContracts() // based on ContractStatus != Terminated
      .filter { it.hasMasterInception }
      .minBy { it.masterInception }.masterInception

    val periodToCheck = max(firstMasterInception?.let {YearMonth.from(it)), YearMonth.now())
```

Reply:
To answer `What would happen if we just did "YearMonth.now()" straight up?` That's where the bug is if I have an insurance that will start in December I will get no bundle discount back from `getBundleDiscountOrNullForPeriod` since I don't have any contracts today.

Reply:
Using Kotlins `max` would be nice but there is no implementation of it for YearMonth which would mean that I would have to get a unix timestamp of the YearMonth and then convert it back to a YearMonth when done comparing